### PR TITLE
Drone demo changes

### PR DESCRIPTION
**Pull request by JeremyAdamHart**
* Added framebuffers to poggles
* Added target argument to texture upload calls (chiefly for cubemaps)
* Changed initializer list arguments to vectors

### CODE REVIEW

**Review by ebiederstadt — APPROVED** [review]

File: include/poggles/framebuffer.h [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+private:
+  GLenum m_original_target;
+  framebuffer_handle m_framebuffer_handle;
+
+  // TODO: Shared pointer doesn't feel ideal here, consider other object
+  // TODO: These should be linked together in some way
+  std::vector<GLenum> m_drawBuffers;
+};
+}  // namespace poggles
```
Comment (benjaminulmer) [inline include/poggles/framebuffer.h]: \>:

File: source/poggles/framebuffer.cpp [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        std::cout << "GL_FRAMEBUFFER_UNSUPPORTED" << std::endl;
+        break;
+      case GL_FRAMEBUFFER_INCOMPLETE_MULTISAMPLE:
+        std::cout << "GL_FRAMEBUFFER_INCOMPLETE_MULTISAMPLE" << std::endl;
+        break;
+      case GL_FRAMEBUFFER_INCOMPLETE_LAYER_TARGETS:
+        std::cout << "GL_FRAMEBUFFER_INCOMPLETE_LAYER_TARGETS" << std::endl;
+        break;
+    }
+  }
```
Comment (benjaminulmer) [inline source/poggles/framebuffer.cpp]: Do we not have a better way of handling/reporting errors in poggles? (genuine question)

**Review by benjaminulmer — COMMENTED** [review]
The change from initializer lists to vectors makes sense. Are there other places where we need to make this change as well?

File: source/poggles/framebuffer.cpp (hunk 2 above)
Comment (lakinwecker) [reply source/poggles/framebuffer.cpp]: define "better". I chose to not have poggles  choose or depend on a logger. We could switch to having it use `tl::expected`, but I also wasn't sure if that was a good choice for general GIV usage. However, I don't think this library gets general GIV usage, so I might be designing for a target audience that doesn't exist.

Comment (JeremyAdamHart) [reply source/poggles/framebuffer.cpp]: We use error callbacks in the other libraries, right? Maybe just returning false would be good enough and we could rely on those

Comment (benjaminulmer) [reply source/poggles/framebuffer.cpp]: At the very least just throwing the name into std out isn't very helpful. An actual message along with the name of the error to std error would be my preference.

Comment (benjaminulmer) [reply source/poggles/framebuffer.cpp]: Me and Jeremy discussed this irl. We're going to just not do any reporting. OpenGL error reporting is outside the scope of what poggles does.

**Comment by JeremyAdamHart** [comment]
Following a discussion with Ben, and reading more in depth into OpenGL's framebuffer docs, I've removed the framebuffer class in favor of free functions. The class was needlessly restrictive and could be prone to buggy behaviour if not used exactly how I intended it. This is now more in line with the base OpenGL functions, and is a much lighter wrapper

**Comment by benjaminulmer** [comment]
https://github.com/lakinwecker/poggles/actions/runs/5510000501
^ CI run for this PR

**Review by benjaminulmer — APPROVED** [review]
lgtm pending CI results

**Comment by benjaminulmer** [comment]
Pushed a commit that fixed an issue on windows from a previous PR.
https://github.com/lakinwecker/poggles/actions/runs/5510105023

**Pull request merged**